test/Microsoft.AspNetCore.SignalR.Client.Tests/HttpConnectionTests.ConnectionLifecycle.cs

```diff
@@ -136,13 +137,10 @@ Task OnTransportStart()
                             transport: new TestTransport(onTransportStart: OnTransportStart)),
                         async (connection, closed) =>
                     {
-                        var actual = await Assert.ThrowsAsync<Exception>(() => connection.StartAsync());
-                        Assert.Same(expected, actual);
+                        Assert.True(shouldFail);
+                        await connection.StartAsync();
+                        Assert.False(shouldFail);
```

Comment:
Stick a call counter in `OnTransportStart` and verify that too. Otherwise looks good